### PR TITLE
exclude kmsplugin from readiness probe

### PR DESCRIPTION
**Pull request by zshihang**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
we only need healthz check in gke

**Which issue(s) this PR fixes**:
Fixes # b/144707780

```release-note
NONE
```

### CODE REVIEW

**Comment by shihan9** [comment]
\cc @immutableT

**Comment by logicalhan** [comment]
Why would we commit this to master? https://github.com/kubernetes/kubernetes/pull/85868 should accomplish the same thing. 

This may be desirable to manually cherry pick to prior versions however, when applicable.

**Comment by shihan9** [comment]
> Why would we commit this to master? #85868 should accomplish the same thing.
> 
> This may be desirable to manually cherry pick to prior versions however, when applicable.
#85868 doesn't do the same thing cause kmsplugin check is also installed in readyz.

**Comment by immutableT** [comment]
@logicalhan 

I would like to confirm my understanding of what #85868 will accomplish.
kmsPluginHealthChecks (in the master branch) are [added](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/options/etcd.go#L214) to all [three](https://github.com/kubernetes/apiserver/blob/master/pkg/server/config.go#L383:18) checks.
Would not we still need to exclude kms-plugin checks from /readyz endpoint in the manifest (similarly to what this PR suggests)? 
In other words, should we modify #85868 to include this change?

**Comment by shihan9** [comment]
alex chatted with han offline
